ui/src/main/webapp/picsureui/common/session.js

```diff
@@ -0,0 +1,260 @@
+  };
+
+  var handleNotAuthorizedResponse = function () {
+    try {
+      if (expired()) {
+        history.pushState({}, "", "/psamaui/logout");
+      } else {
+        history.pushState({}, "", "/psamaui/not_authorized");
+      }
+    } catch (e) {
+      console.log("Error determining token expiry");
+      history.pushState({}, "", "/psamaui/not_authorized");
+    }
+  };
+
+  var configureAjax = function () {
+    $.ajaxSetup({
+      headers: { Authorization: "Bearer " + session.token },
+      statusCode: {
+        401: function () {
+          sessionOverrides.handleNotAuthorizedResponse
+            ? sessionOverrides.handleNotAuthorizedResponse()
+            : handleNotAuthorizedResponse();
+        },
+        403: function () {
+          history.pushState({}, "", "/psamaui/not_authorized");
+        },
+      },
+    });
+  };
+
+  var authenticated = function (
+    /*userId,*/ token,
+    username,
+    permissions,
+    acceptedTOS
+  ) {
+    //		session.userId = userId;
+    session.token = token;
+    session.username = username;
+    session.permissions = permissions;
+    session.acceptedTOS = acceptedTOS;
+    sessionStorage.setItem("session", JSON.stringify(session));
+    configureAjax();
+  };
+
+  var updatePrivileges = function (deferred) {
+    var queryTemplateRequest = function () {
+      return $.ajax({
+        url:
+          window.location.origin +
+          "/psama/user/me/queryTemplate/" +
+          settings.applicationIdForBaseQuery,
+        type: "GET",
+        headers: {
+          Authorization:
+            "Bearer " + JSON.parse(sessionStorage.getItem("session")).token,
+        },
+        contentType: "application/json",
+      });
+    };
+    var meRequest = function () {
+      return $.ajax({
+        url: window.location.origin + "/psama/user/me",
+        type: "GET",
+        headers: {
+          Authorization:
+            "Bearer " + JSON.parse(sessionStorage.getItem("session")).token,
+        },
+        contentType: "application/json",
+      });
+    };
+    var userRequest = function () {
+      return $.ajax({
+        url: window.location.origin + "/psama/user",
+        type: "GET",
+        headers: {
+          Authorization:
+            "Bearer " + JSON.parse(sessionStorage.getItem("session")).token,
+        },
+        contentType: "application/json",
+      });
+    };
+    $.when(queryTemplateRequest(), meRequest(), userRequest()).then(
+      function (queryTemplateResponse, meResponse, userResponse) {
+        var currentSession = JSON.parse(sessionStorage.getItem("session"));
+        currentSession.queryTemplate = queryTemplateResponse[0].queryTemplate;
+        // Check for data admin role from users
+        var meUUID = meResponse[0].uuid;
+        var meUserIndex = userResponse[0].map((u) => u.uuid).indexOf(meUUID);
+        var meUser = userResponse[0][meUserIndex];
+        if (meUser.roles.map((r) => r.name).indexOf("Data Admin") != -1) {
```

Comment:
This should be a privilege, created using a SQL script here: https://github.com/hms-dbmi/pic-sure-gic-institution/tree/main/custom-migrations/auth